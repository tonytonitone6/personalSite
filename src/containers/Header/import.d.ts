declare module '*.jpg' {
  const value: any
  export default value
}

declare module '*.png' {
  const content: any
  export default content
}

declare module '*.svg' {
  // import React = require('react');
  const value: any
  export const ReactComponent: any
  export default value
}
