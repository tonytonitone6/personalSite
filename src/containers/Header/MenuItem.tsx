import React, {FunctionComponent, useState} from 'react'

interface MenuListType {
  message: string
  item: string
}

const MenuItem: FunctionComponent = (): JSX.Element => {
  return <div>123</div>
}

export default MenuItem
