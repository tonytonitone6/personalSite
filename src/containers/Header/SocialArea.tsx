import React, { memo } from 'react';

import {
  SocialIcon,
  SocialArea
} from './styles';


interface TypeProps {
  href: string;
  item: string;
}

const SocialAreaComponent = (props: { skillList:  TypeProps[]}) => {
  const menuList = props.skillList;

  const onRenderSocialList = (item: TypeProps) => {
    return (
      <SocialIcon 
        key={item.href}
        target="_blank"
        href={item.href}
      >
        <i className={item.item} />
      </SocialIcon>
    )
  }
  
  return (
    <SocialArea>
      { menuList.map(onRenderSocialList) }
    </SocialArea>
    )
  }
  
  export const MemoSocialArea = memo(SocialAreaComponent)
