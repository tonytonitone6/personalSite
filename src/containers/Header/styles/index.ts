import styled, { keyframes } from 'styled-components';
import bgPhotos from '../../../images/header.jpg';

const moveToRight = keyframes`
  0% {
    transform: translateX(-100px);
    opacity: 0;
  }

  100% {
    transform: translateX(0);
    opacity: 1;
  }
`;


export const HeaderWrapper = styled.header`
  height: 100%;
`;

export const HeaderContainer = styled.div`
  position: relative;
  height: 100%;
  background-image: linear-gradient(rgba(100, 100, 100, 0.5), rgba(100, 100, 100, 0.5)), url(${bgPhotos});
  background-size: cover;
  background-position: center;
  display: flex;
  justify-content: flex-end;
`;

export const HeaderMenu = styled.div`
  @media (max-width: 576px) {
    display: none;
  }

  & > ul {
    text-transform: uppercase;

    &> li {
      display: inline-block;
      margin: 1rem 1rem 0 1rem;
      position: relative;
      
      &::after {
        content: '';
        display: inline-block;
        background-color: rgba(1, 4, 29, 1);
        height: .1rem;
        width: 0;
        transition: all .3s;
        position: absolute;
        left: 0;
        bottom: -0.8rem;
      }

      &:hover::after {
        width: 100%;
      }

      & > a {
        text-decoration: none;
        color: #fff;
      }
    }
  }
`;

export const HeaderSocial = styled.div`
  position: absolute;
  bottom: 0;
  height: 30%;
  width: 100%;
  background: #000;
  opacity: 0.9;
`;

export const Introduction = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);  
  & > h1 {
    animation: ${moveToRight} 1s ease-in;
    text-align: center;
  }
  & > span {
    font-size: 3rem;
    display: block;
    margin: 2rem;
    text-align: center;
  }
  @media (max-width: 576px) {
    & > span {
      font-size: 0.8rem;
    }
  }
  @media (max-width: 1024px) {
    & > span {
      font-size: 1rem;
    }
  }
`;

export const SocialArea = styled.div`
  display: flex;
  justify-content: center;
`;

export const SocialIcon = styled.a`
  margin: 1rem 2rem;
  & > [class*="fab"] {
    color: #fff;
    font-size: 2rem;
    transition: all 1s;
    &:hover {
      color: red;
    }
  }
`;