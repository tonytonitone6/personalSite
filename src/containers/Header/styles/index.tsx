import styled, { keyframes } from 'styled-components';
import bgPhotos from '../../../images/header.jpg';

const moveToRight = keyframes`
  0% {
    transform: translateX(-100px);
    opacity: 0;
  }

  100% {
    transform: translateX(0);
    opacity: 1;
  }
`;

const moveToDown = keyframes`
  0% {
    opacity: 0;
    transform: translateY(4rem);
  }

  100% {
    opacity: 1;
    transform: none;
  }
`;

const pulse = keyframes`
  0% {
    opacity: 1;
    transform: none;
  }

  50% {
    opacity: .8;
    transform: scale(0.8);
  }

  100% {
    opacity: 1;
    transform: scale(1.2);
  }

`;


export const HeaderWrapper = styled.header`
  height: 100%;
`;

export const HeaderContainer = styled.div`
  position: relative;
  height: 100%;
  background-image: linear-gradient(rgba(55, 93, 100, 0.5), rgba(226, 235, 229, 0.5)), url(${bgPhotos});
  background-size: cover;
  background-position: center;
  display: flex;
  justify-content: flex-end;
`;

export const HeaderMenu = styled.div`
  @media (max-width: 576px) {
    display: none;
  }

  position: fixed;
  z-index: 10;
  & > ul {
    text-transform: uppercase;

    &> li {
      display: inline-block;
      margin: 1rem 1rem 0 1rem;
      position: relative;

      &::after {
        content: '';
        display: inline-block;
        background-color: rgba(1, 4, 29, 1);
        height: .1rem;
        width: 0;
        transition: all .3s;
        position: absolute;
        left: 0;
        bottom: -0.8rem;
      }

      &:hover::after {
        width: 100%;
      }

      & > a {
        text-decoration: none;
        color: #000;
      }
    }
  }
`;

export const HeaderSocial = styled.div`
  position: absolute;
  bottom: 0;
  height: 30%;
  width: 100%;
  background: #000;
  opacity: 0.9;
`;

export const Introduction = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  height: 50%;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);  
  & > h1 {
    animation: ${moveToRight} 1s ease-in;
    text-align: center;
  }
  & > span {
    font-size: 3rem;
    display: block;
    margin: 2rem;
    text-align: center;
  }
  @media (max-width: 576px) {
    & > span {
      font-size: 0.8rem;
    }
  }
  @media (max-width: 1024px) {
    & > span {
      font-size: 1rem;
    }
  }
`;

export const SocialArea = styled.div`
  display: flex;
  justify-content: center;
`;

export const SocialIcon = styled.a`
  margin: 1rem 2rem;
  & > [class*="fab"] {
    color: #fff;
    font-size: 2rem;
    transition: all 1s;
    &:hover {
      color: red;
    }
  }
`;


export const ArrowIcon = styled.img`
  position: absolute;
  width: 1.8rem;
  height: 1.8rem;
  bottom: 3rem;
  left: 0;
  right: 0;
  margin: auto;
  animation: ${moveToDown} 1s 1s ease-out forwards, ${pulse} 2s 3s ease-out infinite;
  opacity: 0;
  cursor: pointer;
`