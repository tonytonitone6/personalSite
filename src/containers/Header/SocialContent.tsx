import React, { FunctionComponent, useState, useMemo } from 'react';
import {MemoSocialArea} from './SocialArea';
import {useMenuValue} from '@context/index'
import {
  PersonInfo,
  Avator,
  ArrowIcon
} from './styles';

import ArrowLogo from '../../images/down-arrow.png'

const SocialContent: FunctionComponent = () => {
  const [skillList, setSkillList] = useState([
    {href: 'https://twitter.com/stanmao', item: 'fab fa-twitter-square'},
    {href: 'https://www.facebook.com/yuhsaing.mao', item: 'fab fa-facebook'},
    {
      href: 'https://www.instagram.com/tonytonitone6/?hl=zh-tw',
      item: 'fab fa-instagram',
    },
    {href: 'https://github.com/tonytonitone6', item: 'fab fa-github'},
  ]);

  const [{refs}, ignoreDispatch] = useMenuValue();

  const moveToNextPage = (): void => {
    refs[0].current.scrollIntoView({
      behavior: 'smooth',
      block: 'start',
    })
  }

  const memoSkillList = useMemo(() => skillList, []);


  return (
    <PersonInfo>
      <h1>Stan Mao</h1>
      <p>back end Engineer</p>
      <Avator />
      <MemoSocialArea skillList={memoSkillList} />
      <ArrowIcon 
        src={ArrowLogo} 
        onClick={moveToNextPage}
      />
    </PersonInfo>
  )
}


export default SocialContent;