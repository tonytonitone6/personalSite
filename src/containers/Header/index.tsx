import React, {FunctionComponent, useState, MouseEvent, useMemo} from 'react'
import {FormattedMessage} from 'react-intl'

import {MemoSocialArea} from './SocialArea'
import {useMenuValue} from '@context/index'
import types from '@reducers/constants'

import ArrowLogo from '../../images/down-arrow.png'
import {
  HeaderWrapper,
  HeaderContainer,
  HeaderMenu,
  Introduction,
  ArrowIcon,
} from './styles'

interface TypeMenuItem {
  id: number
  name: string
}

const Header: FunctionComponent = () => {
  const [skillList, setSkillList] = useState([
    {href: 'https://twitter.com/stanmao', item: 'fab fa-twitter-square'},
    {href: 'https://www.facebook.com/yuhsaing.mao', item: 'fab fa-facebook'},
    {
      href: 'https://www.instagram.com/tonytonitone6/?hl=zh-tw',
      item: 'fab fa-instagram',
    },
    {href: 'https://github.com/tonytonitone6', item: 'fab fa-github'},
  ])

  const [{menuList, refs}, _] = useMenuValue()

  const handleClick = (e: MouseEvent<HTMLLIElement>, id: number) => {
    e.preventDefault()
    refs[id].current.scrollIntoView({
      behavior: 'smooth',
      block: 'start',
    })
  }

  const onRenderMenuList = ({
    id,
    name,
  }: {
    id: number
    name: string
  }): JSX.Element | null => {
    if (menuList && menuList.length !== 0) {
      return (
        <li key={id} onClick={e => handleClick(e, id)}>
          <a href="#">
            <FormattedMessage id={name} />
          </a>
        </li>
      )
    }
    return null
  }

  const moveToNextPage = (): void => {
    refs[0].current.scrollIntoView({
      behavior: 'smooth',
      block: 'start',
    })
  }

  return (
    <HeaderWrapper>
      <HeaderContainer>
        <HeaderMenu>
          <ul>
            {menuList && menuList.length !== 0
              ? menuList.map(onRenderMenuList)
              : null}
          </ul>
        </HeaderMenu>
        <Introduction>
          <h1>Stan Mao</h1>
          <span>Backend Engineer</span>
          <MemoSocialArea skillList={useMemo(() => skillList, [])} />
        </Introduction>
        <ArrowIcon src={ArrowLogo} onClick={moveToNextPage} />
      </HeaderContainer>
    </HeaderWrapper>
  )
}

export default Header
