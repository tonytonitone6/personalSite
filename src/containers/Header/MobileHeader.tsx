import React, {FunctionComponent} from 'react'

import {NavBar, NavBarIcon} from './styles'

interface Iprops {
  active: boolean
  onToggle: () => void
}

const MobileHeader: FunctionComponent<Iprops> = ({
  active,
  onToggle,
}): JSX.Element => {
  return (
    <NavBar active={active} onClick={onToggle}>
      <NavBarIcon />
      <NavBarIcon />
      <NavBarIcon />
    </NavBar>
  )
}

export default MobileHeader
