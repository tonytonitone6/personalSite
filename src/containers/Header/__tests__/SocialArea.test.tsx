import React from 'react';
import {render} from '@testing-library/react';

import {MemoSocialArea} from '@containers/Header/SocialArea';
import {
  SocialArea,
  SocialIcon
} from '../styles';

import 'jest-styled-components';

const defaultProps = {
  skillList: [
    {href: 'https://twitter.com/stanmao', item: 'fab fa-twitter-square'},
    {href: 'https://www.facebook.com/yuhsaing.mao', item: 'fab fa-facebook'},
    {
      href: 'https://www.instagram.com/tonytonitone6/?hl=zh-tw',
      item: 'fab fa-instagram',
    },
    {href: 'https://github.com/tonytonitone6', item: 'fab fa-github'}
  ],
}

const initialComponent = (props = {}) => {
  const settingProps = {...props}
  return render(<MemoSocialArea {...settingProps} />)
}

describe('render icon', () => {
  it('should return target icon', () => {
    const {queryByText, debug} = initialComponent(defaultProps)
    const icon = queryByText('https://twitter.com/stanmao')
    // debug(icon);
  })

  it('render icon list', () => {
    const {getByTestId} = initialComponent(defaultProps)
    expect(getByTestId('renders-socialArea').children.length).toBe(defaultProps.skillList.length);
  })

  it('pass zero props to render icon', () => {
    const wrapper = initialComponent({...defaultProps});
    expect(wrapper).toMatchSnapshot();
  })
})
