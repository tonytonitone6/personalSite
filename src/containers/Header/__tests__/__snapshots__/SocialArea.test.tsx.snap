// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render icon pass zero props to render icon 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  margin: 1rem 2rem;
}

.c1 > [class*='fab'] {
  color: #fff;
  font-size: 2rem;
  -webkit-transition: all 1s;
  transition: all 1s;
}

.c1 > [class*='fab']:hover {
  color: red;
}

<body>
    <div>
      <div
        class="c0"
        data-testid="renders-socialArea"
      >
        <a
          class="c1"
          data-testid="test-icon"
          href="https://twitter.com/stanmao"
          target="_blank"
        >
          <i
            class="fab fa-twitter-square"
          />
        </a>
        <a
          class="c1"
          data-testid="test-icon"
          href="https://www.facebook.com/yuhsaing.mao"
          target="_blank"
        >
          <i
            class="fab fa-facebook"
          />
        </a>
        <a
          class="c1"
          data-testid="test-icon"
          href="https://www.instagram.com/tonytonitone6/?hl=zh-tw"
          target="_blank"
        >
          <i
            class="fab fa-instagram"
          />
        </a>
        <a
          class="c1"
          data-testid="test-icon"
          href="https://github.com/tonytonitone6"
          target="_blank"
        >
          <i
            class="fab fa-github"
          />
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-EHOje hacebR"
      data-testid="renders-socialArea"
    >
      <a
        class="sc-bZQynM ccNtSi"
        data-testid="test-icon"
        href="https://twitter.com/stanmao"
        target="_blank"
      >
        <i
          class="fab fa-twitter-square"
        />
      </a>
      <a
        class="sc-bZQynM ccNtSi"
        data-testid="test-icon"
        href="https://www.facebook.com/yuhsaing.mao"
        target="_blank"
      >
        <i
          class="fab fa-facebook"
        />
      </a>
      <a
        class="sc-bZQynM ccNtSi"
        data-testid="test-icon"
        href="https://www.instagram.com/tonytonitone6/?hl=zh-tw"
        target="_blank"
      >
        <i
          class="fab fa-instagram"
        />
      </a>
      <a
        class="sc-bZQynM ccNtSi"
        data-testid="test-icon"
        href="https://github.com/tonytonitone6"
        target="_blank"
      >
        <i
          class="fab fa-github"
        />
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
